spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageProtocolResolver.java

```diff
@@ -36,21 +36,26 @@
  * @author Vinicius Carvalho
  * @author Artem Bilan
  */
-public class GoogleStorageProtocolResolver implements ProtocolResolver, BeanFactoryPostProcessor, ResourceLoaderAware {
+public class GoogleStorageProtocolResolver
+		implements ProtocolResolver, BeanFactoryPostProcessor, ResourceLoaderAware {
 
 	private static final Log logger = LogFactory.getLog(GoogleStorageProtocolResolver.class);
 
 	private static final String PROTOCOL = "gs://";
 
 	private ConfigurableListableBeanFactory beanFactory;
 
+	private GoogleStorageProtocolResolverSettings settings
+			= new GoogleStorageProtocolResolverSettings();
```

Comment:
Why create the setting here?

Reply:
This is a *default* state. It's absolutely OK to have it during object initialization.
If end-user decides to have its own option (s)he is always free to declare such a bean in the application context. Exactly this happens when we use Spring Boot.

Not sure what is your concern: this default is here for those users who don't use Boot and don't want to override something.

Reply:
OK, thanks for the clarification. I missed the fact that we don't fail when the settings bean doesn't exist.